src/lib.rs

```diff
@@ -1 +1,192 @@
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct Child {
+    process_information: PROCESS_INFORMATION,
+}
+
+impl Child {
+    fn new(
+        command: &OsStr,
+        inherit_handles: bool,
+        current_directory: Option<&PathBuf>,
+    ) -> Result<Self> {
+        unsafe {
+            let mut si = STARTUPINFOW::default();
+            let mut pi = PROCESS_INFORMATION::default();
+
+            si.cb = size_of::<STARTUPINFOW>() as u32;
+
+            let process_creation_flags = PROCESS_CREATION_FLAGS(0);
+
+            let res = if let Some(directory) = current_directory {
+                let directory = directory.as_os_str();
+                windows::Win32::System::Threading::CreateProcessW(
+                    PWSTR::default(),
+                    command,
+                    std::ptr::null() as *const SECURITY_ATTRIBUTES,
+                    std::ptr::null() as *const SECURITY_ATTRIBUTES,
+                    inherit_handles,
+                    process_creation_flags,
+                    std::ptr::null() as *const c_void,
+                    directory,
+                    &si,
+                    &mut pi as *mut PROCESS_INFORMATION,
+                )
+            } else {
+                windows::Win32::System::Threading::CreateProcessW(
+                    PWSTR::default(),
+                    command,
+                    std::ptr::null() as *const SECURITY_ATTRIBUTES,
+                    std::ptr::null() as *const SECURITY_ATTRIBUTES,
+                    inherit_handles,
+                    process_creation_flags,
+                    std::ptr::null() as *const c_void,
+                    PWSTR::default(),
+                    &si,
+                    &mut pi as *mut PROCESS_INFORMATION,
+                )
+            };
+
+            if res.as_bool() {
+                Ok(Self {
+                    process_information: pi,
+                })
+            } else {
+                Err(Error::CreationFailed(get_last_error()))
+            }
+        }
+    }
+
+    pub fn wait(&self) -> ExitStatus {
+        let mut exit_code: u32 = 0;
+
+        unsafe {
+            WaitForSingleObject(self.process_information.hProcess, INFINITE);
+            GetExitCodeProcess(
+                self.process_information.hProcess,
+                &mut exit_code as *mut u32,
+            );
+
+            CloseHandle(self.process_information.hProcess);
+            CloseHandle(self.process_information.hThread);
+        }
+
+        ExitStatus(exit_code)
+    }
+
+    pub fn try_wait(&self) -> Result<Option<ExitStatus>> {
+        let mut exit_code: u32 = 0;
+
+        let res = unsafe {
+            GetExitCodeProcess(
+                self.process_information.hProcess,
+                &mut exit_code as *mut u32,
+            )
+        };
+
+        if res.as_bool() {
+            match exit_code {
+                259 => Ok(None),
+                _ => Ok(Some(ExitStatus(exit_code))),
```

Comment:
After a successful try_wait you probably need to call the CloseHandle functions to emulate properly the behavior of std::process. This is because if a try_wait succeeded, the user don't need to call `wait` to reap off the process.